src/storage/s3.rs

```diff
@@ -100,9 +100,13 @@ impl<'a> S3Backend<'a> {
     }
 }
 
-fn parse_timespec(raw: &str) -> Result<Timespec, Error> {
-    const TIME_FMT: &str = "%a, %d %b %Y %H:%M:%S %Z";
-    Ok(time::strptime(raw, TIME_FMT)?.to_timespec())
+fn parse_timespec(mut raw: &str) -> Result<DateTime<Utc>, Error> {
+    raw = raw.trim_end_matches(" GMT");
+
+    Ok(DateTime::from_utc(
+        NaiveDateTime::parse_from_str(raw, "%a, %d %b %Y %H:%M:%S")?,
+        Utc,
+    ))
```

Comment:
Can you add a test for this?

Reply:
There already are some [here](https://github.com/rust-lang/docs.rs/pull/781/files/8ef89945beaf1de95a589228e7e980188526fa1d#diff-c3a2a09475e9683a0e82b9a1d6f5e0b9R148-R156), do you want more?

Reply:
Those look great, thanks.